server_addon/houdini/client/ayon_houdini/plugins/load/load_hda.py

```diff
@@ -18,38 +39,44 @@ class HdaLoader(plugin.HoudiniLoader):
     color = "orange"
 
     def load(self, context, name=None, namespace=None, data=None):
-        import hou
 
         # Format file name, Houdini only wants forward slashes
         file_path = self.filepath_from_context(context)
         file_path = os.path.normpath(file_path)
         file_path = file_path.replace("\\", "/")
 
-        # Get the root node
-        obj = hou.node("/obj")
-
         # Create a unique name
         counter = 1
         namespace = namespace or context["folder"]["name"]
         formatted = "{}_{}".format(namespace, name) if namespace else name
         node_name = "{0}_{1:03d}".format(formatted, counter)
 
+        hda_defs = hou.hda.definitionsInFile(file_path)
+        if not hda_defs:
+            raise RuntimeError ("No HDA definitions found!")
+
+        hda_def = hda_defs[0]
+        parent_node = self._create_dedicated_parent_node(hda_def)
+
         hou.hda.installFile(file_path)
-        hda_node = obj.createNode(name, node_name)
+        hda_node = parent_node.createNode(name, node_name)
+        hda_node.moveToGoodPosition()
 
-        self[:] = [hda_node]
+        # Imprint it manually
+        data = {
+            "schema": "openpype:container-2.0",
+            "id": AVALON_CONTAINER_ID,
+            "name": node_name,
+            "namespace": namespace,
+            "loader": str(self.__class__.__name__),
```

Comment:
```suggestion
            "loader": self.__class__.__name__,
```

Reply:
good catch! 
It's wired that `str` casting is used in other places in AYON addons although it's already a string.
Thanks!

Reply:
May have been some odd legacy in Py2 where there may somehow have been a difference between `str` and `unicode`, etc?